torch/lib/c10d/TCPStore.cpp

```diff
@@ -286,8 +287,10 @@ TCPStore::TCPStore(
     tcpStoreDaemon_ = std::unique_ptr<TCPStoreDaemon>(
         new TCPStoreDaemon(masterListenSocket_));
   }
+  // set the timeout
+  setTimeout(timeout);
```

Comment:
Ideally this is set directly in the base class constructor. Can you add a constructor to `Store.hpp` that takes the timeout and sets it? We can keep the existing default constructor that initializes with the default timeout.